api/src/services/SolrIndexer.ts

```diff
@@ -295,47 +295,55 @@ class SolrIndexer {
 
         fields.has_order_str = ((fields["relsext.sortOn_txt_mv"] ?? [])[0] ?? "title") === "custom" ? "yes" : "no";
 
-        for (const field in fedoraData.agents) {
+        const agents = await fedoraData.extraDetails.getAgents();
+        for (const field in agents) {
             const fieldName = "agent." + field + "_txt_mv";
-            fields[fieldName] = fedoraData.agents[field];
+            fields[fieldName] = agents[field];
         }
 
-        if (fedoraData.license !== null) {
-            fields["license.mdRef_str"] = fedoraData.license;
+        const license = await fedoraData.getLicense();
+        if (license !== null) {
+            fields["license.mdRef_str"] = license;
```

Comment:
I think you can do null coalescing for all the code below. `fields["license.mdRef_str"] = await fedoraData.getLicense() ?? fields["license.mdRef_str"];`

Reply:
My goal here is to only set `fields["license.mdRef_str"]` to a value if there is a non-null license URL retrieved, and to have the key remain undefined in other situations. Since `fields["license.mdRef_str"]` is not defined prior to the assignment, wouldn't your proposed code result in an error related to accessing an undefined object property and/or result in a key being set to a null value instead of completely omitted? (Javascript is not my primary language, so perhaps it just behaves differently than I assume it behaves -- if that's the case, I apologize).

Reply:
Oh okay.